### PR TITLE
Framework cleanup

### PR DESCRIPTION
**Pull request by hgy59**
## Description

Some pending fixes and cleanups on my TODO list

## Checklist

- [x] Avoid icons in packages that do not integrate into DSM UI
- [x] Avoid overwriting build logs for successive make calls
- [x] Enable patches for noarch and native builds
- [x] Avoid irrelevant warnings when building wheels for python2
- [x] Consistently use BUILD_DEPENDS for native dependencies
       This triggers a lot of packages to build, but without code changes.
- [x] Use explicit PKG_DIST_FILE when PKG_DIST_NAME does not contain the package name
- [x] Shorten some github archive urls
- [x] Improve cross/x265 code generation
       This is the only code change in this PR - regarding ffmpeg (and dependents) and imagemagick packages

### Type of change

- [x] Small framework changes

### CODE REVIEW

File: cross/ffmpeg/Makefile [hunk 1]
```diff
@@ -183,17 +181,19 @@ CONFIGURE_ARGS += --cpu=e500 --disable-altivec
 endif
 endif
 
+
+ifeq ($(findstring $(ARCH),$(i686_ARCHS) $(x64_ARCHS)),$(ARCH))
+BUILD_DEPENDS = native/nasm
+NASM_PATH = $(relapath $(WORK_DIR)/../../../native/nasm/work-native/install/usr/local/bin)
+ENV += PATH=$(NASM_PATH):$$PATH
+endif
```
Comment (th0ma7) [inline cross/ffmpeg/Makefile:189]: Have you found out why it failed for x64-6.1 and 7.0?

File: mk/spksrc.wheel.mk [hunk 2]
```diff
@@ -85,9 +85,11 @@ ifneq ($(strip $(WHEELS)),)
 	      crossenvPIP=$$(. $(CROSSENV) && which pip) ; \
 	      $(MSG) "Python crossenv found: [$(CROSSENV)]" ; \
 	      $(MSG) "pip crossenv found: [$${crossenvPIP}]" ; \
-	   else \
+	   elif [ "$(PYTHON_VERSION)" != "2.7" ] ; then \
 	      $(MSG) "WARNING: Python crossenv NOT found!" ; \
 	      $(MSG) "WARNING: pip crossenv NOT found!" ; \
+	   else \
+	      $(MSG) "Python $(PYTHON_VERSION) uses pip: $${crossenvPIP}" ; \
```
Comment (th0ma7) [inline mk/spksrc.wheel.mk:92]: As an FYI, I intent to revisit python 2.7 in a short future.  In the meantime (and perhaps also in the long-run) this change looks good.

**Review by th0ma7 — CHANGES_REQUESTED** [review]
It currently fails on x64 which needs investigating.  Other than that all changes looks good.

File: cross/ffmpeg/Makefile (hunk 1 above)
Comment (hgy59) [reply cross/ffmpeg/Makefile]: finally yes (just a typo)

File: mk/spksrc.wheel.mk (hunk 2 above)
Comment (hgy59) [reply mk/spksrc.wheel.mk:92]: @th0ma7 I have python2 in progress for #5148.
As python2 spk contains some wheels that have vulnerabilities and are not updated for python 2 anymore, I am about to remove most of the wheels in that package (keep only pip, virtualenv and setuptools).
And probably we should publish python2 for DSM 7 too, as python2 by Synology does not contain pip, virtualenv and setuptools.

Comment (th0ma7) [reply mk/spksrc.wheel.mk:92]: I'm good with that, and then we can publish ffsync

**Pull request merged**